apps/overload/src/app/vehicle.utils.ts

```diff
@@ -28,28 +28,40 @@ interface Boat {
 
 type Vehicle = Bicycle | Car | Moto | Bus | Boat;
 
+export function createVehicle(type: 'bicycle'): Bicycle;
+export function createVehicle(type: 'car', fuel: Fuel): Car;
+export function createVehicle(type: 'moto', fuel: Fuel): Moto;
+export function createVehicle(
+  type: 'boat',
+  capacity: number,
+  isPublicTransport: boolean
```

Comment:
this `isPublicTransport` flag should be set in the 'bus' function declaration and removed from the boat one.